pulsar-client/src/main/java/org/apache/pulsar/client/api/MessageId.java

```diff
@@ -37,6 +36,16 @@
      */
     byte[] toByteArray();
 
+    /**
+     * Get the topic name of this MessageId.
+     * This is mainly for TopicsConsumerImpl to identify a message belongs to which topic.
+     *
+     * @return the topic name
+     */
+    default String getTopicName() {
```

Comment:
I understand that the implemenation needs to know which topic originated the message so that it can "acknowledge" on the right internal consumer, but I think this doesn't need to be exposed in the `MessageId` interface.

Reply:
Yes. It is as @merlimat metioned. for MessageId, we also need `topicName` for `redeliverUnacknowledgedMessages`